enumerables.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+    def my_each
+        if block_given?
+            item = 0
+            while item < self.size
+                yield(self[item])
+             item+=1
+            end
+        end
+    
+    self
+    end
+
+
+
+    def my_each_with_index
+        if block_given?
+            index = 0
+            while index < self.size
+                yield(self[index], index)
+                index+=1
+            end
+        end
+        self
+    end
+
+    def my_select
+        if block_given?
+            filtered = []
+            index = 0
+            self.my_each { |x| filtered.push(x) if yield(x) }
+        end
+        filtered
+    end
+    
+    def my_all?
+        if block_given?
+            not_all = 0
+            index = 0
+            while index < self.size
+                not_all = 1 if !yield(self[index])
+                index+=1
+                break if not_all == 1
+            end
+            
+        end
+        not_all == 1 ? false : true
+    end
+
+    def my_any?
+        if block_given?
+            any = 0
+            index = 0
+            while index < self.size
+                any = 1 if yield(self[index])
+                index+=1
+                break if any == 1
+            end
+        end
+        any == 1 ? true : false
+    end
```

Comment:
Please see the comment on `my_all?`

### Find more details about [`any?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F)